pkg/providers/mac_keychain_darwin.go

```diff
@@ -8,3 +8,188 @@ import (
 	"github.com/spectralops/teller/pkg/core"
 	"github.com/spectralops/teller/pkg/logging"
 )
+
+// pathPrefix is the prefix for the keychain path
+const pathPrefix = "teller-"
```

Comment:
When I initially tested it, it didn't work. However, during the debugging process, I discovered that you have restricted the retrieval of only keys with the prefix "teller-" for added security, which is a commendable approach.

To ensure clarity and consistency, I recommend documenting this restriction in both the README.md file and the metaInfo section of the init() function. By doing so, other users and developers will be aware of this requirement and can adhere to it when working with the codebase.